modulemd/v2/tests/test-modulemd-profile.c

```diff
@@ -80,6 +80,111 @@ profile_test_construct (ProfileFixture *fixture, gconstpointer user_data)
 }
 
 
+static void
+profile_test_equals (ProfileFixture *fixture, gconstpointer user_data)
+{
+  g_autoptr (ModulemdProfile) p_1 = NULL;
+  g_autoptr (ModulemdProfile) p_2 = NULL;
+  g_auto (GStrv) rpms;
+
+  /*Test 2 objects with same name*/
+  p_1 = modulemd_profile_new ("testprofile");
+  g_assert_nonnull (p_1);
+  g_assert_true (MODULEMD_IS_PROFILE (p_1));
+
+  p_2 = modulemd_profile_new ("testprofile");
+  g_assert_nonnull (p_2);
+  g_assert_true (MODULEMD_IS_PROFILE (p_2));
+
+  g_assert_true (modulemd_profile_equals (p_1, p_2));
+
+  g_clear_object (&p_1);
+  g_clear_object (&p_2);
+
+  /*Test 2 objects with different name*/
+  p_1 = modulemd_profile_new ("testing");
+  g_assert_nonnull (p_1);
+  g_assert_true (MODULEMD_IS_PROFILE (p_1));
+
+  p_2 = modulemd_profile_new ("testprofile");
+  g_assert_nonnull (p_2);
+  g_assert_true (MODULEMD_IS_PROFILE (p_2));
+
+  g_assert_false (modulemd_profile_equals (p_1, p_2));
+
+  g_clear_object (&p_1);
+  g_clear_object (&p_2);
+
+  /* Test 2 profile objects with same name and description */
+  p_1 = modulemd_profile_new ("testprofile");
+  modulemd_profile_set_description (p_1, "a test");
+  g_assert_nonnull (p_1);
+  g_assert_true (MODULEMD_IS_PROFILE (p_1));
+
+  p_2 = modulemd_profile_new ("testprofile");
+  modulemd_profile_set_description (p_2, "a test");
+  g_assert_nonnull (p_2);
+  g_assert_true (MODULEMD_IS_PROFILE (p_2));
+
+  g_assert_true (modulemd_profile_equals (p_1, p_2));
+
+  g_clear_object (&p_1);
+  g_clear_object (&p_2);
+
+  /* Test 2 profile objects with same name and different description */
+  p_1 = modulemd_profile_new ("testprofile");
+  modulemd_profile_set_description (p_1, "a test");
+  g_assert_nonnull (p_1);
+  g_assert_true (MODULEMD_IS_PROFILE (p_1));
+
+  p_2 = modulemd_profile_new ("testprofile");
+  modulemd_profile_set_description (p_2, "b test");
+  g_assert_nonnull (p_2);
+  g_assert_true (MODULEMD_IS_PROFILE (p_2));
+
+  g_assert_false (modulemd_profile_equals (p_1, p_2));
+
+  g_clear_object (&p_1);
+  g_clear_object (&p_2);
+
+  /* Test 2 profile objects with same name, description, and rpms */
+  p_1 = modulemd_profile_new ("testprofile");
+  modulemd_profile_set_description (p_1, "a test");
+  modulemd_profile_add_rpm (p_1, "testrpm");
+  g_assert_nonnull (p_1);
+  g_assert_true (MODULEMD_IS_PROFILE (p_1));
+
+  p_2 = modulemd_profile_new ("testprofile");
+  modulemd_profile_set_description (p_2, "a test");
+  modulemd_profile_add_rpm (p_2, "testrpm");
+  g_assert_nonnull (p_2);
+  g_assert_true (MODULEMD_IS_PROFILE (p_2));
+
+  g_assert_true (modulemd_profile_equals (p_1, p_2));
+
+  g_clear_object (&p_1);
+  g_clear_object (&p_2);
+
+  /* Test 2 profile objects with same name, description, and different rpms */
```

Comment:
Please add two more tests:

* Compare two RPM sets where the first sorted value matches and the second does not. e.g. `[ "a", "b" ]` vs `[ "a", "c" ]`.
* Compare two RPM sets where the first sorted value matches, but one has more entries than the other. e.g. `[ "a", "b" ]` vs `[  "a", "b", "c" ]`.

The first one will make sure the code doesn't accidentally return TRUE if the first one seen matches. The second one makes sure we return TRUE if they are different size sets.

Reply:
> The second one makes sure we return TRUE if they are different size sets.

You mean FALSE, right?

Reply:
Oops, yes I definitely meant FALSE.